internal/observability/middleware.go

```diff
@@ -105,3 +105,21 @@ func (m *Middleware) CallOneway(ctx context.Context, req *transport.Request, out
 	call.End(err, false /* isApplicationError */)
 	return ack, err
 }
+
+// HandleStream implements middleware.StreamInbound.
+func (m *Middleware) HandleStream(serverStream transport.ServerStream, h transport.StreamHandler) error {
+	call := m.graph.begin(serverStream.Context(), transport.Streaming, true /* isInbound */, serverStream.Request().Meta.ToRequest())
```

Comment:
use temporaries instead of comments, or change the underlying API to be enum-like?

Reply:
(same below)

Reply:
seems reasonable, will change.